### ISSUE TITLE
[Main] Downloads for anonfiles fails with 

### ISSUE THREAD

**Issue by OzzieIsaacs** [pyload/pyload#3756]
### Description

I tried to download a file from anonfiles e.g. `https://anonfiles.com/v5a2adX7oa/Europa-Report.part1_rar`.
After appling the patch PR #3753 the download fails with the following traceback. I tried to debug it, got different errors. To me it looks like the download gets started several times in parallel.

I'm now done, this is to time consuming for me. I stop testing and improving pyload NG for now.

### Debug log
```
[2020-09-20 07:34:50]  INFO                pyload  Download starts: Europa-Report (13).part1_rar
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_preparing`
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON UserAgentSwitcher: Setting connection timeout to 60 seconds
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON UserAgentSwitcher: Setting maximum redirections to 10
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON UserAgentSwitcher: Use custom user-agent string `Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0`
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: Plugin version: 0.52
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: Plugin status: testing
[2020-09-20 07:34:50]  WARNING             pyload  DOWNLOADER DefaultPlugin[9]: Plugin may be unstable
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: Config option `use_premium` not found, use default `True`
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Grabbing link info...
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: Link info: {'name': 'Europa-Report.part1_rar', 'hash': {}, 'pattern': {}, 'size': 0, 'status': 7, 'url': 'https://anonfiles.com/v5a2adX7oa/Europa-Report.part1_rar'}
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: Previous link info: {}
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Link name: Europa-Report.part1_rar
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Link size: N/D
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Link status: starting
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Processing url: https://anonfiles.com/v5a2adX7oa/Europa-Report.part1_rar
[2020-09-20 07:34:50]  DEBUG               pyload  DOWNLOADER DefaultPlugin[9]: DOWNLOAD URL https://anonfiles.com/v5a2adX7oa/Europa-Report.part1_rar | get={} | post={} | ref=False | cookies=True | disposition=True | resume=None | chunks=None | fixurl=True
[2020-09-20 07:34:50]  DEBUG               pyload  Chunk 1 download finished
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: File saved
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: Checking download...
[2020-09-20 07:34:50]  INFO                pyload  DOWNLOADER DefaultPlugin[9]: File is OK
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_processed`
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `package_processed`
[2020-09-20 07:34:50]  INFO                pyload  Download finished: Europa-Report.part1_rar
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_finished`
[2020-09-20 07:34:50]  INFO                pyload  Package finished: ano
Traceback (most recent call last):
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/threads/download_thread.py", line 62, in run
    self.pyload.files.check_package_finished(pyfile)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/managers/file_manager.py", line 603, in check_package_finished
    self.pyload.addon_manager.package_finished(pyfile.package())
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/utils/old/__init__.py", line 144, in wrapper
    return func(self, *args, **kwargs)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/managers/addon_manager.py", line 240, in package_finished
    plugin.package_finished(package)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  [Previous line repeated 987 more times]
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 198, in package_finished
    def package_finished(self, pypack):
RecursionError: maximum recursion depth exceeded while calling a Python object
[2020-09-20 07:34:50]  WARNING             pyload  Download failed: Europa-Report.part1_rar | maximum recursion depth exceeded while calling a Python object
Traceback (most recent call last):
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/threads/download_thread.py", line 62, in run
    self.pyload.files.check_package_finished(pyfile)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/managers/file_manager.py", line 603, in check_package_finished
    self.pyload.addon_manager.package_finished(pyfile.package())
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/utils/old/__init__.py", line 144, in wrapper
    return func(self, *args, **kwargs)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/core/managers/addon_manager.py", line 240, in package_finished
    plugin.package_finished(package)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 199, in package_finished
    return self.package_finished(pypack)
  [Previous line repeated 987 more times]
  File "/home/ozzie/.local/lib/python3.8/site-packages/pyload/plugins/base/addon.py", line 198, in package_finished
    def package_finished(self, pypack):
RecursionError: maximum recursion depth exceeded while calling a Python object
[2020-09-20 07:34:50]  DEBUG               pyload  Error creating zip file: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/tmp/pyLoad/DefaultPlugin'
[2020-09-20 07:34:50]  INFO                pyload  Debug Report written to /tmp/pyLoad/debug_DefaultPlugin_2020-09-20_07-34-50.txt
[2020-09-20 07:34:50]  INFO                pyload  ADDON UnSkipOnFail: Looking for skipped duplicates of: Europa-Report.part1_rar (pid:9)
[2020-09-20 07:34:50]  INFO                pyload  ADDON UnSkipOnFail: No duplicates found
[2020-09-20 07:34:50]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_failed`
```
### Additional references

Tested on Linux Mint 20 with pyhton 3.8
